build/gen_stub.php

```diff
@@ -1480,23 +1483,380 @@ private function getFlagsAsArginfoString(): string
+    private function getParameterSection(DOMDocument $doc): DOMElement {
+        $parametersRefSec = $this->generateRefSect1($doc, 'parameters');
+        if (empty($this->args)) {
+            $noParamEntity = $doc->createEntityReference('no.function.parameters');
+            $parametersRefSec->appendChild($noParamEntity);
+            return $parametersRefSec;
+        } else {
+            $parametersPara = $doc->createElement('para');
+            $parametersRefSec->appendChild($parametersPara);
+
+            $parametersPara->appendChild(new DOMText("\n   "));
+            $parametersList = $doc->createElement('variablelist');
+            $parametersPara->appendChild($parametersList);
+
+            /*
+            <varlistentry>
+             <term><parameter>name</parameter></term>
+             <listitem>
+              <para>
+               Description.
+              </para>
+             </listitem>
+            </varlistentry>
+            */
+            foreach ($this->args as $arg) {
+                $parameter = $doc->createElement('parameter', $arg->name);
+                $parameterTerm = $doc->createElement('term');
+                $parameterTerm->appendChild($parameter);
 
-        return $doc->saveXML();
+                $parameterEntry = $doc->createElement('varlistentry');
+                $parameterEntry->appendChild(new DOMText("\n     "));
+                $parameterEntry->appendChild($parameterTerm);
+                $parameterEntry->appendChild(new DOMText("\n     "));
+
+                $listItemPara = $doc->createElement('para');
+                $listItemPara->appendChild(new DOMText("\n       "));
+                $listItemPara->appendChild(new DOMText("Description."));
+                $listItemPara->appendChild(new DOMText("\n      "));
+
+                $parameterEntryListItem = $doc->createElement('listitem');
+                $parameterEntryListItem->appendChild(new DOMText("\n      "));
+                $parameterEntryListItem->appendChild($listItemPara);
+                $parameterEntryListItem->appendChild(new DOMText("\n     "));
+
+                $parameterEntry->appendChild($parameterEntryListItem);
+                $parameterEntry->appendChild(new DOMText("\n    "));
+
+                $parametersList->appendChild(new DOMText("\n    "));
+                $parametersList->appendChild($parameterEntry);
+            }
+            $parametersList->appendChild(new DOMText("\n   "));
+        }
+        $parametersPara->appendChild(new DOMText("\n  "));
+        $parametersRefSec->appendChild(new DOMText("\n "));
+        return $parametersRefSec;
+    }
+
+    private function getReturnValueSection(DOMDocument $doc): DOMElement {
+        $returnRefSec = $this->generateRefSect1($doc, 'returnvalues');
+
+        $returnDescriptionPara = $doc->createElement('para');
+        $returnDescriptionPara->appendChild(new DOMText("\n   "));
+
+        $returnType = $this->return->type;
```

Comment:
I believe this should be:
```suggestion
        $returnType = $this->return->getMethodSynopsisType();
```

(although in practice it doesn't really matter, since `void`, `true`, and `bool` is basically always a native return type.

Reply:
I have no idea how ``$phpDocType`` looks like, and I was expecting this to return me XML markup

Reply:
`$phpDocType` is exactly the same as a regular type, the only difference between them is that the former is parsed from PHPDoc tags and as such, supports some extended types (like array<int, string>)